### DISCUSSION TITLE
Electron tasks

### DISCUSSION THREAD

**Discussion by jgadsden** [OWASP/threat-dragon#363] [General] [answered]
We have various tasks needed for the electron feature for V2.0 Threat Dragon.
This is the discussion area for that

**Comment by lreading** [comment]
Do we want to wrap the server code inside the electron app?

There would be some work getting that working correctly, and it may add some complexity with the integrations (using localhost, SSL certs, etc).

Originally I was thinking we just have a "local only" mode for the electron wrap (local vue session) and disable the other options.  I can see an argument for wrapping all the things in the electron app though.  Open for discussion!

**Comment by ShubhamPalriwala** [comment]
Let's have an environment variable in our env. Using that,
- One case can spins up the backend with our app,
- The other case uses our hosted backend URL when we have it?

**Reply by lreading** [reply]
https://github.com/OWASP/threat-dragon/blob/main/td.vue/src/service/env.js#L4 

^ This feels a bit hacky, I linked the discussion where I got the implementation from.  I haven't tested it yet, but we can presumably use this as a starting point.  I'm not opposed to an environment variable either! It does get a bit tricky though, as node env vars don't exist in the browser.

**Comment by lreading** [comment, marked as answer]
We need electron-specific logic for saving the threat model to disk: #383
